pkg/transport/generic_transport_controller.go

```diff
@@ -146,40 +211,60 @@ func getGvrFromWrappedObject(clientset kubernetes.Interface, wrappedObject runti
 	return restMapping.Resource, nil
 }
 
+// clusterProperties holds the (name, value) pairs that are the properties
+// of a given WEC, for input to customization.
+type clusterProperties = map[string]string
+
 type genericTransportController struct {
 	logger logr.Logger
 
+	inventoryInformerSynced     cache.InformerSynced
+	inventoryLister             clusterlisters.ManagedClusterLister
+	bindingClient               controlclient.BindingInterface
 	bindingLister               controlv1alpha1listers.BindingLister
 	bindingInformerSynced       cache.InformerSynced
+	itsNSClient                 corev1client.NamespaceInterface
+	propCfgMapLister            corev1listers.ConfigMapNamespaceLister
+	propCfgMapInformerSynced    cache.InformerSynced
 	wrappedObjectInformerSynced cache.InformerSynced
 
-	// workqueue is a rate limited work queue.
+	// workqueue is a rate limited work queue of references to objects to work on.
 	// This is used to queue work to be processed instead of performing it as soon as a change happens.
 	// This means we can ensure we only process a fixed amount of resources at a time, and makes it
 	// easy to ensure we are never processing the same item simultaneously in two different workers.
+	// An item can be either of two types: a string holding the name of a Binding, or a
+	// recollectProperties holding the name of a inventory object.
 	workqueue workqueue.RateLimitingInterface
 
 	transport        Transport         //transport is a specific implementation for the transport interface.
 	transportClient  dynamic.Interface // dynamic client to transport wrapped object. since object kind is unknown during complilation, we use dynamic
 	wrappedObjectGVR schema.GroupVersionResource
 
-	wdsClientset     ksclientset.Interface
 	wdsDynamicClient dynamic.Interface
 	wdsName          string
+
+	propsMutex sync.Mutex
+
+	// bindingSensitiveDestinations maps Binding name to the set of destinations whose properties the Binding is senstive to.
+	// Access to both the map and the Sets it holds is controlled by the RWMutex.
+	// The sets are mutable with the RWMutex held.
+	bindingSensitiveDestinations map[string]sets.Set[v1alpha1.Destination]
```

Comment:
Why not use utils.ConcurrentMap?

Reply:
It's an option to expand the interface implementation with a map that shares a mutex with another. In all cases this is not a blocker as it is a personal preference, I can just PR it later if you approve.